app/src/test/java/org/fossasia/openevent/general/event/EventUtilsTest.kt

```diff
@@ -109,4 +112,109 @@ class EventUtilsTest {
             """.trimIndent(), EventUtils.getSharableInfo(event, resource))
     }
 
+    @Test
+    fun `should get timezone name`() {
+        val event = getEvent()
+        val localizedDateTime = getLocalizedDate(event.startsAt)
+        assertEquals("""
+           IST
+            """.trimIndent(), EventUtils.getFormattedTimeZone(localizedDateTime))
+    }
+
+    @Test
+    fun `should get formatted time`() {
+        val event = getEvent()
+        val localizedDateTime = getLocalizedDate(event.startsAt)
+        assertEquals("""
+           04:23 PM
+            """.trimIndent(), EventUtils.getFormattedTime(localizedDateTime))
+    }
+
+    @Test
+    fun `should get formatted date and time without year`() {
+        val event = getEvent()
+        val localizedDateTime = getLocalizedDate(event.startsAt)
+        assertEquals("""
+          Monday, Sep 15
+            """.trimIndent(), EventUtils.getFormattedDateWithoutYear(localizedDateTime))
+    }
+
+    @Test
+    fun `should get formatted date short`() {
+        val event = getEvent()
+        val localizedDateTime = getLocalizedDate(event.startsAt)
+        assertEquals("""
+          Mon, Sep 15
+            """.trimIndent(), EventUtils.getFormattedDateShort(localizedDateTime))
+    }
+
+    @Test
+    fun `should get formatted date`() {
+        val event = getEvent()
+        val localizedDateTime = getLocalizedDate(event.startsAt)
+        assertEquals("""
+          Monday, Sep 15, 2008
+            """.trimIndent(), EventUtils.getFormattedDate(localizedDateTime))
+    }
+
+    @Test
+    fun `should get formatted date range when start and end date are not same`() {
+        val event = getEvent()
+        val startsAt = getLocalizedDate(event.startsAt)
+        val endsAt = getLocalizedDate(event.endsAt)
+        assertEquals("""
+         Mon, Sep 15, 04:23 PM
+            """.trimIndent(), EventUtils.getFormattedEventDateTimeRange(startsAt, endsAt))
+    }
+
+    @Test
+    fun `should get formatted date range when start and end date are same`() {
+        val event = getEvent(endsAt = "2008-09-15T15:53:00+05:00")
+        val startsAt = getLocalizedDate(event.startsAt)
+        val endsAt = getLocalizedDate(event.endsAt)
+        assertEquals("""
+          Monday, Sep 15
+            """.trimIndent(), EventUtils.getFormattedEventDateTimeRange(startsAt, endsAt))
+    }
+
+    @Test
+    fun `should get formatted date range when start and end date are not same in event details`() {
+        val event = getEvent()
+        val startsAt = getLocalizedDate(event.startsAt)
+        val endsAt = getLocalizedDate(event.endsAt)
+        assertEquals("""
+          - Fri, Sep 19, 07:55 PM IST
```

Comment:
dash

Reply:
if I add the dash in the activity I will have to put an if else there.
This is the reason why I am returning it from the EventUtils function